src/Translator.php

```diff
@@ -247,7 +248,14 @@ public function transChoice($message, $number, array $parameters = [], $domain =
 				$result = strtr($message, $parameters);
 
 			} else {
-				$result = $this->formatter->choiceFormat($message, (int) $number, $locale, $parameters);
+				if (!$this->formatter instanceof ChoiceMessageFormatterInterface) {
+					$result = $id;
+					if ($this->panel !== NULL) {
+						$this->panel->choiceError(new \Symfony\Component\Translation\Exception\LogicException(sprintf('The formatter "%s" does not support plural translations.', get_class($this->formatter))), $domain);
```

Comment:
I think you missed this: https://github.com/symfony/translation/blob/012ec5a083f4177496df0ddc516d2b509e2ab8a6/Formatter/ChoiceMessageFormatterInterface.php#L17

Reply:
No, I did not. My intention with this pull request was to fix an error, so we can use this library after a `composer update` without any application errors. Fixing the deprecated notices will be a quite big change, since there are new interfaces, some of the old ones are deprecated, some methods in the `Translator` service are also deprecated, which looks like a breaking change to me. The `TranslationExtension` uses also deprecated methods (`setClass`).

Reply:
Ok, that makes sense.